### PR TITLE
[Frontend] The loading state only last for 5 seconds when reporting

### PR DESCRIPTION
**Pull request by badukwei**
## Summary

修改 report flow 與更新相關組件

## Linked Issue

close https://github.com/social-tw/social-tw-website/issues/554

## Details

- Remove the reporting success dialog
- Remove the reporting failure dialog
- 在載入對話框上新增關閉按鈕
- 顯示載入對話框 5 秒並自動關閉對話框

### CODE REVIEW

File: packages/frontend/src/features/post/components/Comment/CommentReportDialog.tsx [hunk 1]
```diff
@@ -64,22 +63,30 @@ export function CommentReportDialog({
...
+
+            const timer = setTimeout(() => {
+                onCloseDialog()
+            }, 5000)
+
+            return () => {
+                clearTimeout(timer)
+            }
+        }
+    }, [isPending, onCloseDialog])
```
Comment (Xiawpohr) [inline packages/frontend/src/features/post/components/Comment/CommentReportDialog.tsx:84]: 如果當前 epoch 已做過 user state transition 時，reportComment 的時間會少於 5 秒，因而導致 5 秒過後不會自動關閉 ReportFormSubmitting。

**Review by Xiawpohr — COMMENTED** [review]
Thanks for your contribution

File: packages/frontend/src/features/post/components/Comment/CommentReportDialog.tsx [hunk 2]
```diff
@@ -109,14 +116,10 @@ export function CommentReportDialog({
...
-            {isError && (
-                <ReportFormSubmitFailure
-                    onClose={onCloseDialog}
-                    onResubmit={onFailureResubmit}
-                />
-            )}
-            {isSuccess && <ReportFormSubmitSuccess onClose={onCloseDialog} />}
+            <ReportFormSubmitting
+                isOpen={isSubmitting}
+                onClose={onCloseDialog}
```
[the remark this answers is not in the document]
Comment (badukwei) [reply packages/frontend/src/features/post/components/Comment/CommentReportDialog.tsx:130]: 已修正

**Review by Xiawpohr — APPROVED** [review]
Approved~

**Pull request merged**